src/app/lib/stillingSoekSchema.ts

```diff
@@ -172,42 +183,40 @@ export const transformed = elasticSearchAdResultSchema.transform(({ _source, _id
 
         // Employer
         employer: getEmployerData(_source),
-        contactList: _source.contactList,
+        contactList: _source?.contactList,
 
         // For debugging
-        categoryList: _source.categoryList,
+        categoryList: _source?.categoryList,
         searchtags: properties?.searchtags,
         education: properties?.education,
         experience: properties?.experience,
         needDriversLicense: properties?.needDriversLicense,
         under18: properties?.under18,
     };
-});
-export type AdDTORAWSchema = z.infer<typeof adDTORAWSchema>;
-export type UrlDTO = z.infer<typeof urlDTOSchema>;
-export type EmployerDTO = z.infer<typeof employerDTOSchema>;
-export type ContactDTO = z.infer<typeof contactDTOSchema>;
-export type LocationDTO = z.infer<typeof locationSchema>;
-export type MappedAdDTO = z.infer<typeof transformed>;
-
+}
 /**
  *  --------------------------- Common Functions ---------------------------
  */
 
-function getUrl(url: string | undefined): UrlDTO | undefined {
-    if (!url) {
+export function getUrl(url: string | undefined): UrlDTO | undefined {
+    if (url == null) {
         return undefined;
     }
+
     try {
-        const validUrl = new URL(url);
-        if (validUrl.protocol.startsWith("http")) {
-            return { url };
+        if (isValidUrl(url)) {
+            // Legg til https:// om protokollen mangler
+            if (!/^https?:\/\//i.test(url)) {
+                url = `https://${url}`;
+            }
+            return { url: url };
+        } else {
+            return { dangerouslyInvalidUrl: url };
```

Comment:
Ser nå at her så kan det havne gyldige url, som ikke er http/https, men f.eks ftp, blob, eller med en annen slags ukjent eller uønsket protokoll. Disse ønsker vi vel heller ikke å returnere videre?

Reply:
Jeg har nå gjort om på isValidUrl, den kaster nå ingen feil, men returnerer true eller false. håndterer også litt flere feil, har satt opp noen tester for dette, se utilsts.ts => isValidUrl og utils.test.ts for hva jeg har gjort, og kom gjerne med input om du ser noe muffens @otenav